arangod/Transaction/Methods.cpp

```diff
@@ -565,8 +565,30 @@ bool transaction::Methods::sortOrs(
     }
 
     auto conditionData = static_cast<ConditionData*>(parts[i].data);
-    root->addMember(conditionData->first);
-    usedIndexes.emplace_back(conditionData->second);
+    bool haveSeen = false;
+
+    if (!usedIndexes.empty()) {
```

Comment:
Won't this always be false? We clear `usedIndexes` above, and as far as I can tell, we only add to it inside this `if`.

Reply:
It is only false in the first iteration of the loop.
At the bottom of the loop we do insert into `usedIndexes`.

Reply:
Sorry, I mismatched my brackets. Comment withdrawn.